src/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramResultFormField/DeclaredProgramResultFormField.test.ts

```diff
@@ -0,0 +1,175 @@
+import type { AddDeclaredProgramForm, DeclaredProgramFormData } from '@/features/student/declaredPrograms/types/forms.types'
+import DeclaredProgramResultFormField from '@/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramResultFormField/DeclaredProgramResultFormField.vue'
+import { DECLARED_PROGRAM_RESULT_MAX_LENGTH } from '@/features/student/declaredPrograms/config'
+import { BddTest } from '@avenirs-esr/avenirs-dsav/test-utils'
+import { useForm } from '@tanstack/vue-form'
+import { mount, type VueWrapper } from '@vue/test-utils'
+import { beforeEach, expect, vi } from 'vitest'
+
+const DeclaredProgramResultInputStub = defineComponent({
+  name: 'DeclaredProgramResultInput',
+  props: {
+    modelValue: String,
+    errorMessage: String
+  },
+  emits: ['update:modelValue', 'blur'],
+  template: `
+    <div data-testid="declared-program-result-input-stub">
+      <input
+        :value="modelValue"
+        @input="$emit('update:modelValue', $event.target.value)"
+        @blur="$emit('blur')"
+      />
+    </div>
+  `
+})
+
+const TestWrapper = defineComponent({
+  components: {
+    DeclaredProgramResultFormField
+  },
+  setup () {
+    const form = useForm({
+      defaultValues: {
+        result: ''
+      } as DeclaredProgramFormData,
+      validators: {
+        onSubmit () {
+          return { fields: { result: undefined } }
+        }
+      }
+    }) as unknown as AddDeclaredProgramForm
+
+    return { form }
+  },
+  template: `
+    <form @submit.prevent="form.handleSubmit">
+      <DeclaredProgramResultFormField :form="form" />
+    </form>
+  `
+})
+
+BddTest().given('a declared program result form field', () => {
+  let wrapper: VueWrapper<InstanceType<typeof TestWrapper>>
+
+  const stubs = {
+    DeclaredProgramResultInput: DeclaredProgramResultInputStub
+  }
+
+  BddTest().when('the component is mounted', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should render the result input component', () => {
+      const input = wrapper.findComponent({ name: 'DeclaredProgramResultInput' })
+      expect(input.exists()).toBe(true)
+    })
+
+    BddTest().then('it should have empty initial value', () => {
+      const input = wrapper.findComponent({ name: 'DeclaredProgramResultInput' })
+      expect(input.props('modelValue')).toBe('')
+    })
+  })
+
+  BddTest().and('the user enters a result', () => {
+    beforeEach(async () => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
```

Comment:
```suggestion
  BddTest().and('the user enters a result', () => {
```

idem que les autres fichiers de test : and dans when et suppression du code répété, valable pour tous les and